src/AutoRest.CSharp/Common/Generation/Writers/CodeWriter.cs

```diff
@@ -356,11 +443,34 @@ public CodeWriter LineRaw(string str)
 
         public CodeWriter AppendRaw(string str) => AppendRaw(str.AsSpan());
 
-        private CodeWriter AppendRaw(ReadOnlySpan<char> span)
+        private CodeWriter AppendRaw(ReadOnlySpan<char> span) => InsertRaw(span, _length);
+
+        private CodeWriter InsertRaw(ReadOnlySpan<char> span, int position)
         {
+            Debug.Assert(0 <= position);
+            Debug.Assert(position <= _length);
+
+            if (Environment.NewLine == "\r\n")
```

Comment:
I'm not sure if this check is correct though. Maybe we should search for "\r\n" regardless of OS

Reply:
I think you are right. Doing it in every OS should be fine.

Reply:
Fixed